### PR TITLE
Add image captioners

### PR DESCRIPTION
**Pull request by hello-cpaxton**
## Description

For integration with OVMM code. Can be tested with:

```bash
# Moonbeam captioner
# Gives: "A plush toy resembling a spotted animal is lying on its back on a wooden floor, with its head and front paws raised."
python -m stretch.perception.captioners.moonbeam_captioner --image_path object.png
# Gives: "An open cardboard box rests on a wooden floor, with a stuffed animal lying next to it."
python -m stretch.perception.captioners.moonbeam_captioner --image_path receptacle.png

# ViT + GPT2 captioner
# Gives: "a cat laying on the floor next to a door"
python -m stretch.perception.captioners.vit_gpt2_captioner --image_path object.png
# Gives: "a box with a cat laying on top of it"
python -m stretch.perception.captioners.vit_gpt2_captioner --image_path receptacle.png

# Blip Captioner
# Gives: "a stuffed dog on the floor"
python -m stretch.perception.captioners.blip_captioner --image_path object.png
# Gives: "a dog laying on the floor next to a cardboard box"
python -m stretch.perception.captioners.blip_captioner --image_path receptacle.png
```

## Checklist

- \[ \] I have performed a self-review of my code
- \[ \] If it is a core feature, I have added thorough tests
- \[ \] I have added documentation for the changes
- \[ \] I have updated the README file if necessary
- \[ \] I have run on hardware if necessary

## Screenshots (if applicable)

Add any relevant screenshots or screen recordings to help reviewers understand the changes.

## Additional context

Add any other context or information about the pull request here.

### CODE REVIEW

File: src/stretch/perception/captioners/moonbeam_captioner.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        self.model = AutoModelForCausalLM.from_pretrained(
+            model_id, trust_remote_code=True, revision=revision
+        )
+        self.tokenizer = AutoTokenizer.from_pretrained(model_id, revision=revision)
+
+    def caption_image(self, image: Union[ndarray, Tensor, Image.Image]):
+        if isinstance(image, Image.Image):
+            pil_image = image
+        else:
```
Comment (hello-atharva) [inline src/stretch/perception/captioners/moonbeam_captioner.py:36]: Docstring

File: src/stretch/perception/captioners/vit_gpt2_captioner.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            "nlpconnect/vit-gpt2-image-captioning"
+        ).to(self._device)
+        self.feature_extractor = ViTImageProcessor.from_pretrained(
+            "nlpconnect/vit-gpt2-image-captioning"
+        )
+        self.tokenizer = AutoTokenizer.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+
+    def caption_image(self, image: Union[ndarray, Tensor]):
+        if isinstance(image, Image.Image):
+            pil_image = image
```
Comment (hello-atharva) [inline src/stretch/perception/captioners/vit_gpt2_captioner.py:54]: Docstring

**Review by hello-atharva — COMMENTED** [review]
Works for me. See two comments below.

File: src/stretch/perception/captioners/moonbeam_captioner.py (hunk 1 above)
Comment (hello-cpaxton) [reply src/stretch/perception/captioners/moonbeam_captioner.py]: done

File: src/stretch/perception/captioners/vit_gpt2_captioner.py (hunk 2 above)
Comment (hello-cpaxton) [reply src/stretch/perception/captioners/vit_gpt2_captioner.py]: done

**Pull request merged**